WcaOnRails/app/models/user.rb

```diff
@@ -327,6 +327,34 @@ def avatar_requires_wca_id
     end
   end
 
+  after_save :changed_delegate_status
+  private def changed_delegate_status
+    user = User.find_by_wca_id(wca_id)
```

Comment:
A couple of comments:

- Not every user (or delegate) has a wca id, so it's not good to assume they do, which you are assuming here.
- When doing a `find_*`, you should either check if it found something, or if you expect it to find something, you should use `find_*!`, so it will throw a clear exception when the thing you're looking for does not exist. In this case, you should have done `User.find_by_wca_id!(wca_id)`. However, as mentioned above, we should not be looking up users by wca_id.
- You're actually already inside of a user instance here, so you can refer to that user by just doing `self`. No need to query the database for the user you already have!

Reply:
Noted! Some newbie mistakes here :)